File: src/main/java/pe/com/test_qsystem/distribution_center/DistributionCenterManager.java
package pe.com.test_qsystem.distribution_center;

import java.util.Set;

public class DistributionCenterManager {
    
    private final Set<DistributionCenter> centers;
    
    public DistributionCenterManager(Set<DistributionCenter> centers) {
        this.centers = centers;
    }
    
    public void operate(String center) {
        centers.stream()
                .filter(c -> c.support(center))
                .findFirst()
                .ifPresentOrElse(
                        c -> c.operate(center), 
                        () -> System.out.println("Not found implementation for " + center + " center.")
                );
    }
    
}

File: src/main/kotlin/pe/com/qsystem/test_qsystem/distribution_center/DistributionCenterManager.kt
package pe.com.qsystem.test_qsystem.distribution_center

import java.lang.RuntimeException

class DistributionCenterManager(
    private val centers: Set<DistributionCenter>
) : DistributionCenter {

    override fun support(center: String) = true

    override fun operate(center: String) {
        centers.find { it.support(center) }
            ?.operate(center) ?: throw RuntimeException("Not found $center distribution center implementation.")
    }
}